Ical.Net/Evaluation/Evaluator.cs

```diff
@@ -43,9 +43,10 @@ protected void IncrementDate(ref CalDateTime dt, RecurrencePattern pattern, int
                 case FrequencyType.Yearly:
                     dt = old.AddDays(-old.DayOfYear + 1).AddYears(interval);
                     break;
-                // FIXME: use a more specific exception.
                 default:
-                    throw new Exception("FrequencyType.NONE cannot be evaluated. Please specify a FrequencyType before evaluating the recurrence.");
+                    // Frequency should always be valid at this stage.
+                    System.Diagnostics.Debug.Fail($"'{pattern.Frequency}' as RecurrencePattern.Frequency is not implemented.");
```

Comment:
Not sure we should fail here. Users could easily set any int as frequency type, which would not be an internal error. I think throwing was not wrong.

Reply:
... kike with `(FrequencyType) 123` - yes, you're right. Wouldn't checking for valid values in the setter of `RecurrencePattern.Frequency` be better?